File: Unity_Example/Scripts/cubecollision.js
#pragma strict


function OnCollisionEnter (col : Collision){
  renderer.material.color = Color.white;
  transform.parent = null;
  transform.rotation = Quaternion.Euler(0, 0, 0);
    if(col.gameObject.name == "Beach"){
      transform.position.y = col.gameObject.transform.position.y + transform.localScale.y/2 + .1;
      rigidbody.isKinematic = true;
      rigidbody.useGravity = true;
    }else {
    
    rigidbody.useGravity = true;
    } 
}

function OnCollisionStay() {
		// if the cubes land in a colliding position, apply force until they move apart.
	   rigidbody.AddForce(transform.forward * 20);
	
	}

File: examples/unity/Scripts/Cubecollision.cs
using UnityEngine;
using System.Collections;

public class Cubecollision : MonoBehaviour {

	void OnCollisionEnter (Collision col){
		GetComponent<Renderer>().material.color = Color.white;
		transform.parent = null;
		transform.rotation = Quaternion.Euler(0, 0, 0);
		if(col.gameObject.name == "Beach"){
			Vector3 newPos = transform.position;
			newPos.y = col.gameObject.transform.position.y + transform.localScale.y/2.0f + .1f;
			transform.position = newPos;

			GetComponent<Rigidbody>().isKinematic = true;
			GetComponent<Rigidbody>().useGravity = true;
		}else {
			
			GetComponent<Rigidbody>().useGravity = true;
		} 
	}
	
	void OnCollisionStay() {
		// if the cubes land in a colliding position, apply force until they move apart.
		GetComponent<Rigidbody>().AddForce(transform.forward * 20);
		
	}
}
